# ESERCITAZIONE FINALE MODULO M4 - PYTHON

### 1. IMPORTO IL DATASET COVID-19 CON PANDAS

In [8]:
import pandas as pd
dtypes={"tests_units":object}
covid19_data = pd.read_csv("\\Users\gabri\Downloads\covid_data.csv", encoding="latin1", low_memory=False)

### 2. VERIFICO LE DIMENSIONI DEL DATASET E LE LE DICITURE PRESENTI NELL'INTESTAZIONE

In [9]:
print("| DIMENSIONI DATASET COVID-19 |")
covid19_data.shape

| DIMENSIONI DATASET COVID-19 |


(311453, 67)

In [10]:
covid19_data.head(20)

iso_code continent     location        date  total_cases  new_cases  \
0       AFG      Asia  Afghanistan  2020-01-03          NaN        0.0   
1       AFG      Asia  Afghanistan  2020-01-04          NaN        0.0   
2       AFG      Asia  Afghanistan  2020-01-05          NaN        0.0   
3       AFG      Asia  Afghanistan  2020-01-06          NaN        0.0   
4       AFG      Asia  Afghanistan  2020-01-07          NaN        0.0   
5       AFG      Asia  Afghanistan  2020-01-08          NaN        0.0   
6       AFG      Asia  Afghanistan  2020-01-09          NaN        0.0   
7       AFG      Asia  Afghanistan  2020-01-10          NaN        0.0   
8       AFG      Asia  Afghanistan  2020-01-11          NaN        0.0   
9       AFG      Asia  Afghanistan  2020-01-12          NaN        0.0   
10      AFG      Asia  Afghanistan  2020-01-13          NaN        0.0   
11      AFG      Asia  Afghanistan  2020-01-14          NaN        0.0   
12      AFG      Asia  Afghanistan  2020-01-15          NaN        0.0   
13      AFG      Asia  Afghanistan  2020-01-16          NaN        0.0   
14      AFG      Asia  Afghanistan  2020-01-17          NaN        0.0   
15      AFG      Asia  Afghanistan  2020-01-18          NaN        0.0   
16      AFG      Asia  Afghanistan  2020-01-19          NaN        0.0   
17      AFG      Asia  Afghanistan  2020-01-20          NaN        0.0   
18      AFG      Asia  Afghanistan  2020-01-21          NaN        0.0   
19      AFG      Asia  Afghanistan  2020-01-22          NaN        0.0   

    new_cases_smoothed  total_deaths  new_deaths  new_deaths_smoothed  ...  \
0                  NaN           NaN         0.0                  NaN  ...   
1                  NaN           NaN         0.0                  NaN  ...   
2                  NaN           NaN         0.0                  NaN  ...   
3                  NaN           NaN         0.0                  NaN  ...   
4                  NaN           NaN         0.0                  NaN  ...   
5                  0.0           NaN         0.0                  0.0  ...   
6                  0.0           NaN         0.0                  0.0  ...   
7                  0.0           NaN         0.0                  0.0  ...   
8                  0.0           NaN         0.0                  0.0  ...   
9                  0.0           NaN         0.0                  0.0  ...   
10                 0.0           NaN         0.0                  0.0  ...   
11                 0.0           NaN         0.0                  0.0  ...   
12                 0.0           NaN         0.0                  0.0  ...   
13                 0.0           NaN         0.0                  0.0  ...   
14                 0.0           NaN         0.0                  0.0  ...   
15                 0.0           NaN         0.0                  0.0  ...   
16                 0.0           NaN         0.0                  0.0  ...   
17                 0.0           NaN         0.0                  0.0  ...   
18                 0.0           NaN         0.0                  0.0  ...   
19                 0.0           NaN         0.0                  0.0  ...   

    male_smokers  handwashing_facilities  hospital_beds_per_thousand  \
0            NaN                  37.746                         0.5   
1            NaN                  37.746                         0.5   
2            NaN                  37.746                         0.5   
3            NaN                  37.746                         0.5   
4            NaN                  37.746                         0.5   
5            NaN                  37.746                         0.5   
6            NaN                  37.746                         0.5   
7            NaN                  37.746                         0.5   
8            NaN                  37.746                         0.5   
9            NaN                  37.746                         0.5   
10           NaN 

### 3. TOTALE CASI COVID-19 PER CONTINENTE

In [11]:
print("| TOTALE CASI COVID-19 PER CONTINENTE |")
def totcasi(covid19_data, cnt1, cnt2, cnt3, cnt4, cnt5, cnt6):
    cnd_1=covid19_data.loc[:, 'continent'] == cnt1
    cnd_2=covid19_data.loc[:, 'continent'] == cnt2
    cnd_3=covid19_data.loc[:, 'continent'] == cnt3
    cnd_4=covid19_data.loc[:, 'continent'] == cnt4
    cnd_5=covid19_data.loc[:, 'continent'] == cnt5
    cnd_6=covid19_data.loc[:, 'continent'] == cnt6
    
    casicovidnw1=covid19_data.loc[cnd_1, 'new_cases'].sum()
    casicovidnw2=covid19_data.loc[cnd_2, 'new_cases'].sum()
    casicovidnw3=covid19_data.loc[cnd_3, 'new_cases'].sum()
    casicovidnw4=covid19_data.loc[cnd_4, 'new_cases'].sum()
    casicovidnw5=covid19_data.loc[cnd_5, 'new_cases'].sum()
    casicovidnw6=covid19_data.loc[cnd_6, 'new_cases'].sum()

    tot_cas={"CONTINENTE":[cnt1, cnt2, cnt3, cnt4, cnt5, cnt6], "CASI TOTALI":[casicovidnw1, casicovidnw2, casicovidnw3, casicovidnw4, casicovidnw5, casicovidnw6]}
        
    return pd.DataFrame(tot_cas)

totcasi(covid19_data, "Africa", "Asia", "Europe", "North America", "Oceania", "South America")

| TOTALE CASI COVID-19 PER CONTINENTE |


CONTINENTE  CASI TOTALI
0         Africa   13088991.0
1           Asia  297234406.0
2         Europe  249319188.0
3  North America  124175560.0
4        Oceania   14073975.0
5  South America   68568119.0

### 4. CONFRONTO DATI COVID-19 PER CONTINENTE

In [12]:
print("| CONFRONTO DATI COVID-19 PER CONTINENTE |")
def inf_conf(covid19_data, cnt1, cnt2, cnt3, cnt4, cnt5, cnt6):
    cnd_1=covid19_data.loc[:, 'continent'] == cnt1
    cnd_2=covid19_data.loc[:, 'continent'] == cnt2
    cnd_3=covid19_data.loc[:, 'continent'] == cnt3
    cnd_4=covid19_data.loc[:, 'continent'] == cnt4
    cnd_5=covid19_data.loc[:, 'continent'] == cnt5
    cnd_6=covid19_data.loc[:, 'continent'] == cnt6
    
    casicovidtot1=covid19_data.loc[cnd_1, 'total_cases']
    casicovidtot2=covid19_data.loc[cnd_2, 'total_cases']
    casicovidtot3=covid19_data.loc[cnd_3, 'total_cases']
    casicovidtot4=covid19_data.loc[cnd_4, 'total_cases']
    casicovidtot5=covid19_data.loc[cnd_5, 'total_cases']
    casicovidtot6=covid19_data.loc[cnd_6, 'total_cases']
    
    mincov1=casicovidtot1.min()
    mincov2=casicovidtot2.min()
    mincov3=casicovidtot3.min()
    mincov4=casicovidtot4.min()
    mincov5=casicovidtot5.min()
    mincov6=casicovidtot6.min()
    
    casicovidnw1=covid19_data.loc[cnd_1, 'new_cases']
    casicovidnw2=covid19_data.loc[cnd_2, 'new_cases']
    casicovidnw3=covid19_data.loc[cnd_3, 'new_cases']
    casicovidnw4=covid19_data.loc[cnd_4, 'new_cases']
    casicovidnw5=covid19_data.loc[cnd_5, 'new_cases']
    casicovidnw6=covid19_data.loc[cnd_6, 'new_cases']
    
    maxcov1=casicovidnw1.max()
    maxcov2=casicovidnw2.max()
    maxcov3=casicovidnw3.max()
    maxcov4=casicovidnw4.max()
    maxcov5=casicovidnw5.max()
    maxcov6=casicovidnw6.max()
    
    mediacov1=round(casicovidnw1.mean(),2)
    mediacov2=round(casicovidnw2.mean(),2)
    mediacov3=round(casicovidnw3.mean(),2)
    mediacov4=round(casicovidnw4.mean(),2)
    mediacov5=round(casicovidnw5.mean(),2)
    mediacov6=round(casicovidnw6.mean(),2)
    
    tc=covid19_data.groupby('continent')
        
    percentot1=round(casicovidnw1.sum()*100/tc['new_cases'].sum().sum(),2)
    percentot2=round(casicovidnw2.sum()*100/tc['new_cases'].sum().sum(),2)
    percentot3=round(casicovidnw3.sum()*100/tc['new_cases'].sum().sum(),2)
    percentot4=round(casicovidnw4.sum()*100/tc['new_cases'].sum().sum(),2)
    percentot5=round(casicovidnw5.sum()*100/tc['new_cases'].sum().sum(),2)
    percentot6=round(casicovidnw6.sum()*100/tc['new_cases'].sum().sum(),2)
    
    ccov={"CONTINENTE":[cnt1, cnt2, cnt3, cnt4, cnt5, cnt6], "MINIMO":[mincov1, mincov2, mincov3, mincov4, mincov5, mincov6], "MASSIMO":[maxcov1, maxcov2, maxcov3, maxcov4, maxcov5, maxcov6], 
          "MEDIA CASI":[mediacov1, mediacov2, mediacov3, mediacov4, mediacov5, mediacov6], "% INFEZIONI":[percentot1, percentot2, percentot3, percentot4, percentot5, percentot6]}
    
    return pd.DataFrame(ccov)

inf_conf(covid19_data, "Africa", "Asia", "Europe", "North America", "Oceania", "South America")

| CONFRONTO DATI COVID-19 PER CONTINENTE |


CONTINENTE  MINIMO    MASSIMO  MEDIA CASI  % INFEZIONI
0         Africa     1.0    46914.0      186.57         1.71
1           Asia     1.0  6966046.0     5142.11        38.78
2         Europe     1.0   500563.0     3973.78        32.53
3  North America     1.0  1265520.0     2462.48        16.20
4        Oceania     1.0   588813.0      476.44         1.84
5  South America     1.0   298408.0     3983.28         8.95

### 5. STATISTICHE VACCINAZIONI COVID-19 PER CONTINENTE

In [13]:
print("| STATISTICHE VACCINAZIONI COVID-19 PER CONTINENTE |")
def vac_cov (covid19_data, cnt1, cnt2, cnt3, cnt4, cnt5, cnt6):
    cnd_1=covid19_data.loc[:, 'continent'] == cnt1
    cnd_2=covid19_data.loc[:, 'continent'] == cnt2
    cnd_3=covid19_data.loc[:, 'continent'] == cnt3
    cnd_4=covid19_data.loc[:, 'continent'] == cnt4
    cnd_5=covid19_data.loc[:, 'continent'] == cnt5
    cnd_6=covid19_data.loc[:, 'continent'] == cnt6
    
    vacovidtot1=covid19_data.loc[cnd_1, 'new_vaccinations']
    vacovidtot2=covid19_data.loc[cnd_2, 'new_vaccinations']
    vacovidtot3=covid19_data.loc[cnd_3, 'new_vaccinations']
    vacovidtot4=covid19_data.loc[cnd_4, 'new_vaccinations']
    vacovidtot5=covid19_data.loc[cnd_5, 'new_vaccinations']
    vacovidtot6=covid19_data.loc[cnd_6, 'new_vaccinations']
    
    minvac1=vacovidtot1.min()
    minvac2=vacovidtot2.min()
    minvac3=vacovidtot3.min()
    minvac4=vacovidtot4.min()
    minvac5=vacovidtot5.min()
    minvac6=vacovidtot6.min()
    
    vacovidnw1=covid19_data.loc[cnd_1, 'new_vaccinations']
    vacovidnw2=covid19_data.loc[cnd_2, 'new_vaccinations']
    vacovidnw3=covid19_data.loc[cnd_3, 'new_vaccinations']
    vacovidnw4=covid19_data.loc[cnd_4, 'new_vaccinations']
    vacovidnw5=covid19_data.loc[cnd_5, 'new_vaccinations']
    vacovidnw6=covid19_data.loc[cnd_6, 'new_vaccinations']
    
    maxvac1=vacovidnw1.max()
    maxvac2=vacovidnw2.max()
    maxvac3=vacovidnw3.max()
    maxvac4=vacovidnw4.max()
    maxvac5=vacovidnw5.max()
    maxvac6=vacovidnw6.max()
    
    mediavac1=round(vacovidnw1.mean(),2)
    mediavac2=round(vacovidnw2.mean(),2)
    mediavac3=round(vacovidnw3.mean(),2)
    mediavac4=round(vacovidnw4.mean(),2)
    mediavac5=round(vacovidnw5.mean(),2)
    mediavac6=round(vacovidnw6.mean(),2)
    
    tc=covid19_data.groupby('continent')
    
    percentotvac1=round(vacovidnw1.sum()*100/tc['new_vaccinations'].sum().sum(),2)
    percentotvac2=round(vacovidnw2.sum()*100/tc['new_vaccinations'].sum().sum(),2)
    percentotvac3=round(vacovidnw3.sum()*100/tc['new_vaccinations'].sum().sum(),2)
    percentotvac4=round(vacovidnw4.sum()*100/tc['new_vaccinations'].sum().sum(),2)
    percentotvac5=round(vacovidnw5.sum()*100/tc['new_vaccinations'].sum().sum(),2)
    percentotvac6=round(vacovidnw6.sum()*100/tc['new_vaccinations'].sum().sum(),2)
    
    vacov={"CONTINENTE":[cnt1, cnt2, cnt3, cnt4, cnt5, cnt6], "MINIMO":[minvac1, minvac2, minvac3, minvac4, minvac5, minvac6], "MASSIMO":[maxvac1, maxvac2, maxvac3, maxvac4, maxvac5, maxvac6], 
          "MEDIA VACCINAZIONI":[mediavac1, mediavac2, mediavac3, mediavac4, mediavac5, mediavac6], "% VACCINAZIONI":[percentotvac1, percentotvac2, percentotvac3, percentotvac4, percentotvac5, percentotvac6]}
    
    return pd.DataFrame(vacov)

vac_cov(covid19_data, "Africa", "Asia", "Europe", "North America", "Oceania", "South America")

| STATISTICHE VACCINAZIONI COVID-19 PER CONTINENTE |


CONTINENTE  MINIMO     MASSIMO  MEDIA VACCINAZIONI  % VACCINAZIONI
0         Africa     0.0   1866822.0            42203.19            1.08
1           Asia     0.0  24741000.0           510400.66           69.60
2         Europe     0.0   2389472.0            59974.27           11.66
3  North America     0.0   7246123.0           174046.22            9.20
4        Oceania     1.0    351226.0            56859.01            0.69
5  South America     1.0   3976605.0           150713.67            7.77

### 6. CONFRONTO STATISTICHE COVID-19 EUROPA-SUD AMERICA-OCEANIA

### Dall'analisi effettuata sui dati derivanti dall'infezione da Covid-19 nel mondo, in particolare per continenti richiesti dal committente, cioè Europa, Sud America e Oceania, si evince che la maggior parte dei casi riscontrati è stata registrata in Europa, con circa 249 milioni di casi riscontrati, contro i 68 milioni e i 14 milioni rispettivamente di Sud America e Oceania. Questo dato è avvalorato anche dalla percentuale di infezioni, che si attesta sul 32,5% in Europa contro gli 8,95% e 1,84% degli altri due continenti. Un dato fortemente significativo però è quello della media casi, altissimo in Sud America e quasi paragonabile a quello europeo (3973 casi di media contro i 3983 dell'Europa) probabilmente motivato dalla densità di popolazione del Sud America, che è sicuramente molto alta. Dato molto basso invece in Oceania, con una media di 476 casi. Un dato che sicuramente salta all'occhio è quello delle vaccinazioni, dove nonostante in Europa si sia vaccinato di più rispetto a Sud America e Oceania, l'incidenza dell'infezione da Covid-19 è stata allo stesso modo molto alta (11% contro i 7,7% e lo 0,69% di Sud America e Oceania), probabilmente a causa della fortissima circolazione del virus nel vecchio continente.